src/dtls_stream_async.rs

```diff
@@ -0,0 +1,61 @@
+use std::{pin::Pin, task::{Context, Poll}, io, fmt};
+use openssl::ssl::{SslRef};
+use tokio::io::{AsyncRead, AsyncWrite};
+use tokio_openssl::SslStream as AsyncSslStream;
+
+use crate::{DtlsStream, DtlsStreamExt}; 
+
+pub type AsyncDtlsStream<S> = DtlsStream<AsyncSslStream<S>>;
```

Comment:
`AsyncDtlsStream` and `SyncDtlsStream` are the same types until used in `UdpChannel`. Maybe we do not have to create separate type definitions for those?